report_dynamic/wizards/wizard_report_dynamic.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 Sunflower IT <http://sunflowerweb.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, fields, models
+
+
+class WizardReportDynamic(models.TransientModel):
+    """Generate reports in bulk"""
+
+    _name = "wizard.report.dynamic"
+    _description = "Select template for report for record(s)"
+
+    template_id = fields.Many2one(
+        comodel_name="report.dynamic",
+        domain=lambda self: [
```

Comment:
@thomaspaulb The idea is to make this domain computed. The "standard" part of the domain will be the requirements that the selected record is a template, with `model_id == active_model`. Then, we loop on all templates in the database for that model_id. In the computation of the domain we add the "standard" part, and `OR `every `template.condition_domain_global`. Then user can select only a template that it's computed domain satisfied all active_ids. Does this make sense?

Reply:
Yeah, however implementationwise, the easiest is to make an invisible, computed field `matching_template_ids` on the wizard, and then on the XML of the form view, not on the Python field, add a domain of "id, in, matching_template_ids"

Reply:
That works only when matching_template_ids is an (invisible) field  on the form view too.

The only drawback of this method is it can get really slow when the number of matching records is > 10000 or something, in which case you would resort to OCA's `web_domain_field` instead. But for matching templates, I don't expect 10000 :-) just a handful max